### PR TITLE
Build Python package for PyPI

### PR DESCRIPTION
**Pull request by bittner**
This PR introduces a number of additions and changes that make building and deploying a Python package easier and cleaner:

- Load resources using [pkg_resources](https://setuptools.readthedocs.io/en/latest/pkg_resources.html) (from `setuptools`) instead of directly from file system
- Handle missing dependencies at runtime (print error message explaining what to do)
- Add a test suite configuration and first tests (BDD)
- Drop Python 2 for linting, tests and builds

### CODE REVIEW

**Comment by cool-RR** [comment]
I see that tests are passing. Ready to merge?

**Comment by bittner** [comment]
As long as the title has the "WIP:" prefix you should consider it "work in progress", hence not ready to merge. I'll change the title as soon as it's ready to merge.

What is missing is the build and upload logic to PyPI [in the Travis configuration](https://docs.travis-ci.com/user/deployment/pypi/). You'll have to help me with submitting credentials to your Travis account (because builds happen on your Travis account), later.

**Comment by cool-RR** [comment]
Cool. I'm not sure whether GitHub notifies on name changes, so make a
comment when you do.

**Comment by bittner** [comment]
> I'm not a fan of the wordy BDD, but I respect your choice.

We can use [pytest](https://docs.pytest.org/en/latest/) instead, technically it doesn't make a difference. (In fact, I first started to prepare a pytest setup.)

The reason I chose [behave](https://behave.readthedocs.io/en/latest/) is that Gherkin is much more self-explanatory (in the build log output), even for users and developers not practical with writing tests. I feel that also closer to the audience of PythonTurtle. Just in case someone may be interesting in looking into the project closer.

**Comment by cool-RR** [comment]
My preference is pytest. If you strongly prefer bdd, free to stay with that.

**Comment by bittner** [comment]
> My preference is pytest.

Let's change that later. I don't insist, but I'd better get the deployment working first.

**Comment by bittner** [comment]
Oh, make it `pypi.password`.

You need to run this from the project repository, having the project cloned on your computer. It will modify the `.travis.yml` file (in an utterly unpleasant way) and add a section like this to it:

```yaml
pypi:
  password:
    secure: I/UBNQygyE....YxwwP4g=
```

I only need the content of the `secure:` block. This is a public value, no need to keep it secret.

**Comment by bittner** [comment]
Hmmm, the more I think about it ... I guess it has to be `deploy.password`. :confused:

File: .travis.yml [hunk 1]
```diff
@@ -1,36 +1,41 @@
...
     - { stage: lint, python: 3.6, env: TOXENV=pylint }
-    - { stage: test, python: 3.7, dist: xenial, sudo: true }
+
+    - stage: publish
+      script: skip
+      deploy:
+        provider: pypi
+        user: coolRR
+        password:
+          secure: <travis-encrypt-value-here>
```
Comment (bittner) [inline .travis.yml]: I think, the PR is now ready to merge. We'll only need to update this password value later (to make uploading to PyPI work).

File: pythonturtle/shelltoprocess/console.py [hunk 2]
```diff
@@ -1,14 +1,14 @@
 """
 See documentation for the class Console defined here.
 """
-import code
+from code import InteractiveConsole
```
Comment (cool-RR) [inline pythonturtle/shelltoprocess/console.py]: I thought we agreed we both like the long module name better?

**Comment by cool-RR** [comment]
Here's the line:

    secure: fdbhH+rsqfG9WfILlBZXlNNSXZVHrfBH9zWa7kKiUY55GMYKli+u/MzG1i6Ezn53Vg1wL1NTQMm5rFbbUfBE2l5/+S191saBZGApEhQZhulSkG702FCbpuGs1a26Zm+dhjyiRvuNdnPB8BA61uHSpWFzDgpLurGsZB0TyB2hZqv7LFL1eG56mKh70LTIDO5XmDh5/yDEXfc23M2nS7Zwx2xJEM047Fay/yeZSjo+LEYgCoTM/z9TftyxkyuLOGsiRoBMwL7eb7wKJzk79vSAvgZP/fSuAqc8ut1lrk1S9OIO4I9s7Zpi1QneJSGDJu4IHfV3nlge/+tlN8aeLA6GsSy/RrrzueBdtYTpbt34vO5rVfhJwjHIKTQa4GUF0R6gCiQ0JjJiXLC+UAbePZxzAKQWaxJWWpqq8L9e+LlHWpEtuB4jEqncFs6FeTT7mTWz2HMTQvvIVAsI2xFkItY4yCpE3P5+5f6UgqwqFpuKCXWF6ohQ090wTlNxN5cikNSYdPPQj7hypKzWrYcH5z4Z4pU63mnO2fZgTEZomvYKXa9mcdtZcCBhPNJW9WTgxqKSouPcjYc4me9wk4wv4ncJaCPLlxbuW7T/Nv+Xy/HYn7eljRapnu7/XcVQgB6gG4vxbRkkLkr1Ybj0xcR2neAQcxhBBqzfCyQezh210dPGqF8=

File: pythonturtle/shelltoprocess/console.py (hunk 2 above)
Comment (bittner) [reply pythonturtle/shelltoprocess/console.py]: This import fixes a redefinition problem in this module at [lines 74++](https://github.com/bittner/PythonTurtle/blob/master/pythonturtle/shelltoprocess/console.py#L74).

What is this `code` module, by the way? -- Crazy, it's a [standard library module](https://docs.python.org/3/library/code.html). Yikes!

**Pull request merged**